pkg/tsdb/loki/loki.go

```diff
@@ -22,6 +24,11 @@ type Service struct {
 	tracer tracing.Tracer
 }
 
+var (
+	_ backend.QueryDataHandler = (*Service)(nil)
+	_ backend.StreamHandler    = (*Service)(nil)
+)
+
```

Comment:
sorry, could you explain what these two are doing?

Reply:
This is a trick to make sure the service implements both `QueryDataHandler` and `StreamHander`  -- compiler will fail with a clean message if something is missing or wrong.  Since nothing is actually declared, it has no runtime consequences.  In java this would be:
```
Service implements QueryDataHandler, StreamHandler { ...
```